src/crawlers/puppeteer_crawler.js

```diff
@@ -311,9 +311,9 @@ class PuppeteerCrawler {
             await this.isRunningPromise;
         } finally {
             if (this.useSessionPool) {
-                this.sessionPool.teardown();
+                await this.sessionPool.teardown();
             }
-            this.puppeteerPool.destroy();
+            await this.puppeteerPool.destroy();
```

Comment:
Just a note, typically `destroy()` functions are used to destroy an object right away no matter what's is its state, so they should not be async and should succeed regardless of the state of the object. I think we should stick to that convention. E.g. see `socket.destroy()`

Reply:
Okay, sure. I didn't know that, will put it to the original state.

Reply:
@jancurn I don't think that's a fair comparison because `socket` is also a stream, which means you can do stuff `.on('close|error')` whereas without the `.destroy()` being `async`, you'd have no way to do that. The `await` was just missing. To make sure you don't have zombie browsers open after `.run()` finishes.

Reply:
PS: We want to make `EventEmitters` out of all crawlers. (Requested by some clients)